### PR TITLE
Exposed some non-critical internal index information.

### PR DESCRIPTION
**Pull request by nemo83**
As part of our experimentation it would be important to access some internal, non critical ANN index info like:

* metric
* some keys examples
* latent factors dimension

For this reason I've refactored the code to expose such data.

### CODE REVIEW

File: src/main/scala/annoy4s/Annoy.scala [hunk 1]
```diff
@@ -26,8 +26,12 @@ class Annoy[T](
   idToIndex: Map[T, Int],
   indexToId: Seq[T],
   annoyIndex: Pointer,
-  dimension: Int
+  val dimension: Int,
+  val metric: Metric
 ) {
+
+  def indexes = indexToId
```
Comment (pishen) [inline src/main/scala/annoy4s/Annoy.scala]: Should this be called `ids`?

File: src/test/scala/annoy4s/AnnoySpec.scala [hunk 2]
```diff
@@ -23,127 +24,147 @@ import scala.util.Random
...
+  private val hammingInputLines = Seq(
+    "a 1 0 0 0",
+    "b 1 1 0 0",
+    "c 1 1 0 1",
+    "d 0 1 1 1"
+  )
+
+  private def getInputDimension(inputVectors: InputVectors): Int = inputVectors.head.split(" ").length - 1
+
+  private def getIndexes(inputVectors: InputVectors): Seq[String] = inputVectors.map(_.split(" ").head)
```
Comment (pishen) [inline src/test/scala/annoy4s/AnnoySpec.scala]: Should be `getIds`?

File: src/main/scala/annoy4s/Annoy.scala (hunk 1 above)
Comment (nemo83) [reply src/main/scala/annoy4s/Annoy.scala]: Oh yeah. Good point, renaming now

**Comment by pishen** [comment]
LGTM, let's merge

**Comment by nemo83** [comment]
<3 great thanks!

**Comment by nemo83** [comment]
can u cut a release when you have a chance? Thanks! :-)

**Comment by pishen** [comment]
Released as 0.8.0!

**Pull request merged**